### PR TITLE
Add examples of CRD propagation, fix crd propagation guide

### PR DESCRIPTION
**Pull request by roytman**

### CODE REVIEW

**Comment by gyliu513** [comment]
/ok-to-test

Welcome @roytman !

File: docs/federating_crds.md [hunk 1]
```diff
@@ -211,16 +218,22 @@ spec:
...
-    - spec
-    - replicas
+   - spec
+   - data
 ```
 
+Deploy it to the federated host cluster
+
+```shell
+$ kubectl apply  --validate=false   -f ./example/crd/federatedBar.yaml --context=cluster1
```
Comment (gyliu513) [inline docs/federating_crds.md]: You need to remove some space for this command.
```shell
kubectl apply --validate=false -f ./example/crd/federatedBar.yaml --context=cluster1
```

File: docs/federating_crds.md [hunk 2]
```diff
@@ -255,16 +268,57 @@ spec:
   - cluster1
```
Comment (gyliu513) [inline docs/federating_crds.md]: kill this line

File: docs/federating_crds.md [hunk 3]
```diff
@@ -255,16 +268,57 @@ spec:
...
 
 ```
+Deploy it to the cluster that hosts federation
+```shell
+kubectl --validate=false apply -f example/crd/federatedbar_test.yaml  --context=cluster1
```
Comment (gyliu513) [inline docs/federating_crds.md]:
```
kubectl --validate=false apply -f example/crd/federatedbar_test.yaml --context=cluster1
```

Remove a space before `--context=cluster1`

File: docs/federating_crds.md [hunk 4]
```diff
@@ -255,16 +268,57 @@ spec:
...
+federatedbar.federation.example.io/test-crd created
+federatedbaroverride.federation.example.io/test-crd created
+federatedbarplacement.federation.example.io/test-crd created
+``` 
 
-```
+Now we can check instances of our CRD
 
+```
```
Comment (gyliu513) [inline docs/federating_crds.md]: Add `shell` here

File: docs/federating_crds.md [hunk 5]
```diff
@@ -255,16 +268,57 @@ spec:
...
+  labels:
+    app: fed-crd
+  name: test-crd
+  namespace: test-namespace
+  resourceVersion: "14445996"
+  selfLink: /apis/example.io/v1/namespaces/test-namespace/bars/test-crd
+  uid: 49d31994-d36e-11e8-a4f3-fa163e77d0a4
+spec:
+  data: Hello
+
```
Comment (gyliu513) [inline docs/federating_crds.md]: kill this line

File: docs/federating_crds.md [hunk 6]
```diff
@@ -255,16 +268,57 @@ spec:
...
 
 $ kubectl get bars -n test-namespace --context=cluster2
 NAME       AGE
 test-crd   30m
 
+$ kubectl get bars test-crd -n test-namespace --context=cluster2 -oyaml
+
```
Comment (gyliu513) [inline docs/federating_crds.md]: kill this line

File: example/crd/federatedbar_crd.yaml [hunk 7]
```diff
@@ -0,0 +1,106 @@
+#template type
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
```
Comment (gyliu513) [inline example/crd/federatedbar_crd.yaml]: How about s/federatedbar_crd.yaml/federatedbar.yaml

As this file was already under `crd` folder

File: example/crd/bar_crd.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
```
Comment (gyliu513) [inline example/crd/bar_crd.yaml]: How about s/bar_crd.yaml/bar.yaml

As this file was already under `crd` folder

File: example/crd/federatedbar_test.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+metadata:
+  name: test-crd
+  namespace: test-namespace
+spec:
+  template:
+    metadata:
+      labels:
+        app: fed-crd
+    spec:
+      data: "Hello"
```
Comment (gyliu513) [inline example/crd/federatedbar_test.yaml]: I'd like you split this file to three files

```
federatedbar-template.yaml
federatedbar-placement.yaml
federatedbar-override.yaml
```

If you put this in one file, then we will always see the data is `World`, I'd like you apply `federatedbar-override.yaml` after finished the test of `federatedbar-template.yaml` and `federatedbar-placement.yaml`.

File: docs/federating_crds.md [hunk 10]
```diff
@@ -1,14 +1,18 @@
 # Federate a CRD in the target cluster
 
-Handling arbitrary resources is one of the essential use-cases for federation. This example will demonstrate how to configure federation to handle a previously unknown, arbitrary custom resource.
+Handling arbitrary resources is one of the essential use-cases for federation. This example will demonstrate how to 
+configure federation to handle a previously unknown, arbitrary Custom Resource.
+
+**Note:**
+If your Custom resource works with a controller, the controller should be handle separately.
```
[the remark this answers is not in the document]
Comment (roytman) [reply docs/federating_crds.md]: Usually CRD works together with one or more controllers. This example propagate instances of CRD only. Related controllers should be distributed separately.

File: example/crd/federatedbar_crd.yaml (hunk 7 above)
Comment (roytman) [reply example/crd/federatedbar_crd.yaml]: The crd suffix shows that it contains CRD definition, opposite to creation of CRD instances.

File: example/crd/bar_crd.yaml (hunk 8 above)
Comment (roytman) [reply example/crd/bar_crd.yaml]: The crd suffix shows that it contains CRD definition, opposite to creation of CRD instances.

File: example/crd/federatedbar_test.yaml (hunk 9 above)
Comment (roytman) [reply example/crd/federatedbar_test.yaml]: In my tests, I saw `Hello` in one cluster and `World` in another.
One file is easy to observe, so I'd prefer to leave it as a single file.

File: docs/federating_crds.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (pmorie) [reply docs/federating_crds.md]:
```suggestion
This guide addresses only propagating instances of a custom resource. Controllers that service custom resource are not addressed here.
```

File: docs/federating_crds.md [hunk 11]
```diff
@@ -1,14 +1,18 @@
...
+Handling arbitrary resources is one of the essential use-cases for federation. This example will demonstrate how to 
+configure federation to handle a previously unknown, arbitrary custom resource.
+
+**Note:**
+If your Custom resource works with a controller, the controller should be handle separately.  
 
 ### Prerequisites
 
-The federation v2 suppports installation for both cluster-scoped and namespace-scoped. In user guide, we are following cluster-scoped federation and deploying federation controller in the namespace `federation-system`. Also, you will be creating the two clusters i.e. `cluster1` and `cluster2` and federation v2 will be installed on `cluster1`.
+The federation v2 suppports installation for both cluster-scoped and namespace-scoped. In [user guide](./userguide.md), we are following cluster-scoped federation and deploying federation controller in the namespace `federation-system`. Also, you will be creating the two clusters i.e. `cluster1` and `cluster2` and federation v2 will be installed on `cluster1`.
```
Comment (pmorie) [inline docs/federating_crds.md]:
```suggestion
You must have federation deployed according to the [user guide](./userguide.md). The cluster where federation is deployed will be `cluster1`. You must have another cluster available to use as `cluster2` in this guide.
```

File: example/crd/federatedbar_test.yaml (hunk 9 above)
Comment (pmorie) [reply example/crd/federatedbar_test.yaml]: I'm also in favor of leaving things in a single file - it's much easier to share as an example that way.

**Comment by roytman** [comment]
@gyliu513 , @pmorie thanks for the comments. I think I resolved most of them, or provided answers to others. How can we go forward ?

File: docs/federating_crds.md [hunk 12]
```diff
@@ -1,26 +1,27 @@
...
+Handling arbitrary resources is one of the essential use-cases for federation. This example will demonstrate how to 
+configure federation to handle a previously unknown, arbitrary custom resource.
+
+**Note:**
+This guide addresses only propagating instances of a custom resource. Controllers that service custom resource are not addressed here.
 
 ### Prerequisites
 
-The federation v2 suppports installation for both cluster-scoped and namespace-scoped. In user guide, we are following cluster-scoped federation and deploying federation controller in the namespace `federation-system`. Also, you will be creating the two clusters i.e. `cluster1` and `cluster2` and federation v2 will be installed on `cluster1`.
+You must have federation deployed according to the [user guide](./userguide.md). The cluster where federation is deployed will be `cluster1`. You must have another cluster available to use as `cluster2` in this guide.
```
Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

Comment (gyliu513) [inline docs/federating_crds.md]: Ah, using [user guide](./userguide.md) is good here.

File: docs/federating_crds.md [hunk 13]
```diff
@@ -1,14 +1,18 @@
...
+Handling arbitrary resources is one of the essential use-cases for federation. This example will demonstrate how to 
+configure federation to handle a previously unknown, arbitrary Custom Resource.
+
+**Note:**
+If your Custom resource works with a controller, the controller should be handle separately.  
 
 ### Prerequisites
 
-The federation v2 suppports installation for both cluster-scoped and namespace-scoped. In user guide, we are following cluster-scoped federation and deploying federation controller in the namespace `federation-system`. Also, you will be creating the two clusters i.e. `cluster1` and `cluster2` and federation v2 will be installed on `cluster1`.
+The federation v2 suppports installation for both cluster-scoped and namespace-scoped. In [user guide](./userguide.md), we are following cluster-scoped federation and deploying federation controller in the namespace `federation-system`. Also, you will be creating the two clusters i.e. `cluster1` and `cluster2` and federation v2 will be installed on `cluster1`.
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply docs/federating_crds.md]: The updated doc looks good to me.

**Pull request merged**